### ISSUE TITLE
'valid-test/suite-description': override error message

### ISSUE THREAD

**Issue by BourgoisMickael** [lo1tuma/eslint-plugin-mocha#204]
I'd like to be able to override the error messagefor a rule

I'm using `mocha/valid-test-description` like that

```js
config.rules['mocha/valid-test-description'] = ['error', /^should(\W* \W*\w+){4,}/]
```
And the error tells me 

```
  37:5  error  Invalid "it()" description found  mocha/valid-test-description
```

I'd like to define my own error message because people who don't look inside the `.eslintrc.js` and don't know the rule they won't be able to fix the problem since it's not descriptive enough, so I'd like to do something like

```js
const errorMessage = 'The description should start with \'should\' and contains at least 4 other words'
config.rules['mocha/valid-test-description'] = ['error', /^should(\W* \W*\w+){4,}/, null, errorMessage]
```

I checked in [`lib/rule/s/valid-test-description.js`](https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/lib/rules/valid-test-description.js) this feature can be added easily by adding

```js
// Line 15
const errorMessage = context.options[2]

// Line 44
context.report(node, errorMessage || `Invalid "${ callee.name }()" description found.`);
```

And the same can be done for `mocha/valid-suite-description`

Since those rules are the only ones who can have a specific pattern they should be the only ones to have the possibility to define a custom error message since the default message can't help to fix the error.

Another possibility is to print the pattern in the error message but I think it will be ugly and not easy to understand

```js
context.report(node, `Invalid "${ callee.name }()" description found, should match ${ pattern }`);
```

**Comment by lo1tuma** [comment]
Sounds reasonable to me.

I think it would be a good idea to use a similar API as [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax#options). Something like `[ 'error', { pattern: "", message: "" } ]`.

**Comment by BourgoisMickael** [comment]
Should I make a PR or you are doing it ?
